### ISSUE TITLE
Last movie play time keeps increasing

### ISSUE THREAD

**Issue by modem** [aunefyren/wrapperr#92]
Everytime I run Wrapperr, the play time of the last played movie and number of plays keeps increasing. If my last movie was played for 1h, the 1st time I load Wrapperr will show 1 hour, 1 play, the 2nd time I load Wrapper it will say 2 hours, 2 plays and so on.

I have 3 users, I see the issue on 2 users, but not on the other. The only difference I can think of is that the users with the issue played movies in December, while the last time the other user played a movie was in November.

The user that does not show this behavior in movies, is showing it in TV Shows.

**Comment by modem** [comment]
Maybe the cache was corrupted, I see this in the log everytime I refresh the Wrapperr page:
```
2024/01/03 10:09:16 New wrap request.
2024/01/03 10:09:16 Checking Tautulli server 'Tautulli'.
2024/01/03 10:09:16 Getting users from Tautulli server 'Tautulli'.
2024/01/03 10:09:16 Getting stats for: User1 User1 195059 user1@gmail.com
2024/01/03 10:09:16 Cache stage completed for User1.
2024/01/03 10:09:16 Checking Tautulli server 'Tautulli'.
2024/01/03 10:09:16 Date 2023-12-07 from server 'Tautulli' marked as incomplete in cache. Refreshing.
2024/01/03 10:09:16 Date 2023-12-12 from server 'Tautulli' marked as incomplete in cache. Refreshing.
2024/01/03 10:09:16 Date 2023-12-22 from server 'Tautulli' marked as incomplete in cache. Refreshing.
2024/01/03 10:09:16 Date 2023-12-23 from server 'Tautulli' marked as incomplete in cache. Refreshing.
2024/01/03 10:09:16 Date 2023-12-29 from server 'Tautulli' marked as incomplete in cache. Refreshing.
2024/01/03 10:09:16 Tautulli refresh/download stage completed.
2024/01/03 10:09:17 Show buddy retrieved.
```
I deleted the cache file and I don't see the issue anymore, neither these log messages about incomplete cache.

From the logs, I see I started with this incomplete cache message when I updated from v3.2.3 to v3.25. Maybe the cache file got corrupted after the update or there was an change in the cache behavior.

**Comment by aunefyren** [comment]
Hey, I had this issue recently as well during development. I deleted the cache and it resolved it self. The downside is I have no clue what went wrong in the cache file.

I might create a v3.2.3 instance with a cache and update to v3.25 to see if I can recreate the issue.

**Comment by dlhall111** [comment]
@aunefyren this issue seems to be easily reproducible with the right criteria, which is that affected movie (not sure if it happens with TV shows?) has to have been watched on the current date and you have to be using a dynamic timeframe.  For example, I have a user that watched Fight Club early this morning and every time I refresh and load up that user's stats the ONLY item that increases as this bug describes is Fight Club.  No other movies this user watched prior to today are increasing with each reload and other users who haven't watched anything today are also not showing any signs of this issue.

Great project BTW, thanks for making it!  :)

**Comment by dlhall111** [comment]
@aunefyren looks like once you induce this situation it will persist until the cache is recreated.  This morning I brought up the same user as yesterday and refresh a few times, the last movie they watched continue to increase in duration/count on every refresh.  Then I blew away the cache and recreated it and could no longer reproduce the problem.  But if I or some other user watches something today and then I go in load that user and refresh, etc. then it should happen again and again and again until I rebuild the cache.

**Comment by aunefyren** [comment]
> [@aunefyren](https://github.com/aunefyren) looks like once you induce this situation it will persist until the cache is recreated. This morning I brought up the same user as yesterday and refresh a few times, the last movie they watched continue to increase in duration/count on every refresh. Then I blew away the cache and recreated it and could no longer reproduce the problem. But if I or some other user watches something today and then I go in load that user and refresh, etc. then it should happen again and again and again until I rebuild the cache.

Good catch, that might help fix this. And the issue presents itself if dynamic timeframe is enabled and something is watched on the day of the wrap-generation?

**Comment by dlhall111** [comment]
@aunefyren correct.  To get specific, this is the rough timeline of events that led to the latest reproduction:

- Setup Wrapperr brand new, latest version, etc.
- Manually build cache
- Let run for a few days without anyone using it (I don't have it exposed yet, just internal so I'd be the only one accessing)
- Noticed a (newish) user had watched something early yesterday, decided to check their Wrapperr output
- Saw the movie they'd watched that morning in their top 10 with 1 play and 2.5 hour duration
- Remembered this bug (I'd noticed it last year and ended up setting a static time to workaround it) and started refreshing to see if I could still reproduce
- Upon each refresh the duration and count would go up
- Never did anything with the cache, just added my comment above yesterday
- Checked Wrapper again this morning to see if I could still reproduce and I could... only affected item was the movie the user watched yesterday morning when I first reproduced the issue
- Went into the admin panel and trashed/rebuilt the cache
- Confirmed everything looked correct for the user on the next run
- Confirmed that reloading their output over and over did NOT induce the issue (**the user has not watching anything today**)

**Comment by dlhall111** [comment]
@aunefyren also, just after I added my last comment I noticed a user watching a TV show so I just did a quick test and can reproduce the issue with TV Shows as well... each refresh the duration/count goes up.  The count goes up by 1 every time but the duration seems to only go up based on how much they've watched today.  Meaning this user has only been watching the show for a few minutes so the duration seems to go up just a bit whereas the user from yesterday who'd watched a whole movie the duration went up a lot more... seems to be duplicating whatever the watch time for that item is for the day and counting the play as a play regardless of time watched.

**Comment by aunefyren** [comment]
Should be fixed in v3.2.9. Thank you for your help @dlhall111

**Comment by dlhall111** [comment]
@aunefyren just pulled the new version and tested, issue fixed!

Thanks for the fix and the great project!
